spec/actions_manager_spec.coffee

```diff
@@ -383,6 +400,16 @@ describe 'ActionsManager', ->
             it 'does not execute callback', ->
               expect(@callback_spy).to.not.be.called
 
+        context 'when data are given as unstringified object', ->
+          beforeEach ->
+            @data = VALID_ORDER_DATA
+
+          it 'stringifies the data before reporting', ->
+            @run()
+            payload = @sendbeacon_spy.args[0][1]
+            expect(@sendbeacon_spy).to.be.called
```

Comment:
We try to avoid multiple expectations in the same test. Maybe either skip this expectation or extract it to a separate test:

```
context 'when data are given as unstringified object', ->
  beforeEach ->
    @data = VALID_ORDER_DATA
    @run()

  it 'reports the action', ->
    expect(@sendbeacon_spy).to.be.called

  it 'stringifies the data before reporting', ->
    payload = @sendbeacon_spy.args[0][1]
    expect(payload.actions[0].data).to.eql VALID_ORDER_DATA_STRINGIFIED
```